core/src/test/scala/unit/kafka/log/LogValidatorTest.scala

```diff
@@ -1397,9 +1397,8 @@ class LogValidatorTest {
         RecordBatch.MAGIC_VALUE_V0, CompressionType.GZIP, CompressionType.GZIP)
     )
     // if there is a mix of both regular InvalidRecordException and InvalidTimestampException,
-    // InvalidTimestampException is no longer takes precedence. The type of invalidException
-    // is unified as InvalidRecordException
-    assertTrue(e.invalidException.isInstanceOf[InvalidRecordException])
+    // InvalidTimestampException takes precedence
+    assertTrue(e.invalidException.isInstanceOf[InvalidTimestampException])
```

Comment:
Since v2.4, (PR: https://github.com/apache/kafka/pull/7612), we intentionally let `InvalidTimestampException` take precedence. I think we should keep the same behavior as before.